api/src/main/java/org/openmrs/module/idgen/IdgenUtil.java

```diff
@@ -98,4 +101,32 @@ public static long convertFromBase(String s, char[] baseCharacters) {
 		}
 		return contents;
 	}
+	
+	/**
+	 * Convenience method for getting a valid prefix from a parent location with a valid prefix attribute up the tree
+	 * 
+	 * @param location the child location. Normally the location for the current {@link UserContext}
+	 * @return prefix
+	 */
+	public static String getLocationPrefixRecursively(Location location) {
```

Comment:
From an API standpoint we don't care that this method is recursive. You can say that in the JavaDoc if you want:
>`Recursively fetches the next prefix up in the tree of locations.`

You can simply name it `getLocationPrefix(Location)`.